tsl/src/remote/healthcheck.c

```diff
@@ -0,0 +1,304 @@
+/*
+ * This file and its contents are licensed under the Timescale License.
+ * Please see the included NOTICE for copyright information and
+ * LICENSE-TIMESCALE for a copy of the license.
+ */
+#include <postgres.h>
+#include <funcapi.h>
+#include <utils/builtins.h>
+#include <utils/lsyscache.h>
+#include <utils/syscache.h>
+
+#include "healthcheck.h"
+#include "data_node.h"
+#include "dist_commands.h"
+#include "dist_util.h"
+
+/*
+ * Functions and data structures for printing a health check result.
+ */
+enum Anum_show_conn
+{
+	Anum_health_node_name = 1,
+	Anum_health_healthy,
+	Anum_health_in_recovery,
```

Comment:
Likely node being in_recovery should not be considered healthy as well...I guess it won't hurt to have extra details (ui maybe?), since we use it for AN as well